contribs/gnodev/cmd/gnodev/main.go

```diff
@@ -0,0 +1,360 @@
+primarily for realm package development. It automatically loads the example folder and any
+additional specified paths.`,
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execDev(cfg, args, stdio)
+		})
+
+	if err := cmd.ParseAndRun(context.Background(), os.Args[1:]); err != nil {
+		fmt.Fprintf(os.Stderr, "%+v\n", err)
+		os.Exit(1)
+	}
+}
+func (c *devCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.webListenerAddr,
+		"web-listener",
+		defaultDevOptions.webListenerAddr,
+		"web server listening address",
+	)
+
+	fs.BoolVar(
+		&c.minimal,
+		"minimal",
+		defaultDevOptions.verbose,
+		"do not load example folder packages",
+	)
+
+	fs.BoolVar(
+		&c.verbose,
+		"verbose",
+		defaultDevOptions.verbose,
+		"verbose output when deving",
+	)
+
+	fs.BoolVar(
+		&c.noWatch,
+		"no-watch",
+		defaultDevOptions.noWatch,
+		"do not watch for files change",
+	)
+
+}
+
+func execDev(cfg *devCfg, args []string, io commands.IO) error {
+	ctx, cancel := context.WithCancelCause(context.Background())
+	defer cancel(nil)
+
+	// guess root dir
+	gnoroot := gnoenv.RootDir()
+
+	pkgpaths, err := parseArgsPackages(io, args)
+	if err != nil {
+		return fmt.Errorf("unable to parse package paths: %w", err)
+	}
+
+	// XXX: find a good way to export or display node logs
+	noopLogger := tmlog.NewNopLogger()
+
+	// RAWTerm setup
+	rt := gnodev.NewRawTerm()
+	{
+		restore, err := rt.Init()
+		if err != nil {
+			return fmt.Errorf("unable to init raw term for dev: %s", err)
+		}
+		defer restore()
+
+		// correctly format output for terminal
+		io.SetOut(commands.WriteNopCloser(rt))
+
+		// Setup trap signal
+		osm.TrapSignal(func() {
+			restore()
+			cancel(nil)
+		})
+	}
+
+	nodeOut := rt.NamespacedWriter("Node")
+	webOut := rt.NamespacedWriter("GnoWeb")
+	keyOut := rt.NamespacedWriter("KeyPress")
+
+	var dnode *gnodev.Node
+	{
+		var err error
+		// XXX: redirect node output to a file
+		dnode, err = setupDevNode(ctx, noopLogger, cfg, pkgpaths, gnoroot)
+		if err != nil {
+			return err // already formated in setupDevNode
+		}
+		defer dnode.Close()
+	}
+
+	fmt.Fprintf(nodeOut, "Listener: %s\n", dnode.GetRemoteAddress())
+
+	// setup files watcher
+	w, err := setupPkgsWatcher(cfg, dnode.ListPkgs())
+	if err != nil {
+		return fmt.Errorf("unable to watch for files change: %w", err)
+	}
+
+	pathChangeCh := make(chan []string, 1)
+	go func() {
+		defer close(pathChangeCh)
+		if err := handleDebounce(ctx, w, pathChangeCh); err != nil {
+			cancel(err)
+		}
+	}()
+
+	// Gnoweb setup
+	server := setupGnowebServer(cfg, dnode, rt)
+
+	l, err := net.Listen("tcp", cfg.webListenerAddr)
+	if err != nil {
+		return fmt.Errorf("unable to listen to %q: %w", cfg.webListenerAddr, err)
+	}
+
+	go func() {
+		var err error
+		if srvErr := server.Serve(l); srvErr != nil {
+			err = fmt.Errorf("HTTP server stopped with error: %w", srvErr)
+		}
+		cancel(err)
+	}()
+	defer server.Close()
+
+	fmt.Fprintf(webOut, "Listener: %s\n", l.Addr())
```

Comment:
```suggestion
	fmt.Fprintf(webOut, "Listener: http://%s\n", l.Addr())
```

So I can click on it from the terminal, I'm to lazy to copy paste :)